commit 7e9198b80e3bc657ff0bb463daab66305e1b78e1 (refs/changes/87/3187/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-01-10T21:20:34+01:00 (1 year, 9 months ago)
    
    TOOLS-1972 'sdcadm avail' crashes when there are not available updates

diff --git a/lib/cli/do_avail.js b/lib/cli/do_avail.js
index cf6a528..0ef2291 100644
--- a/lib/cli/do_avail.js
+++ b/lib/cli/do_avail.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -102,7 +102,13 @@ Available.prototype.execute = function cExecute(opts, args, cb) {
         }
     ]}, function availCb(err) {
         if (err) {
-            return cb(err);
+            cb(err);
+            return;
+        }
+        if (plan.procs.length === 0) {
+            console.log('Up-to-date.');
+            cb();
+            return;
         }
 
         var rows = [];
@@ -169,7 +175,8 @@ Available.prototype.execute = function cExecute(opts, args, cb) {
                     process.stdout.write(JSON.stringify(k) + '\n');
                 });
             }
-            return cb();
+            cb();
+            return;
         }
 
         var validFieldsMap = {};
@@ -187,7 +194,7 @@ Available.prototype.execute = function cExecute(opts, args, cb) {
             sort: sort,
             validFields: Object.keys(validFieldsMap)
         });
-        return cb();
+        cb();
     });
 };
 
